src/main/java/net/barrage/school/java/ecatalog/app/productSource/ProductSourceService.java

```diff
@@ -0,0 +1,13 @@
+package net.barrage.school.java.ecatalog.app.productSource;
+
+public interface ProductSourceService {
+    /**
+     * Load sources
+     */
+    void loadSources();
```

Comment:
It's insecure to sync all merchants in one shoot. U'd better to introduce per-merchant sync API. e.g. `loadSourcesForMerchant(String merchantSourceName)`

Reply:
Actually ur `reloadMerchant` looks pretty well. Just need to remove `if` checking old flag.